htseq2multx.pl

```diff
@@ -0,0 +1,884 @@
+    my $missing = 0;
+    if($prefix =~ m%(.*/)%)
+      {
+        my $dir = $1;
+        if(!-e $dir)
+          {$missing = 1}
+      }
+    return($missing);
+  }
+
+sub outfilesExist
+  {
+    my $hash     = $_[0];
+    my $files    = $_[1];
+    my $existing = 0;
+
+    foreach my $fqf_index (0..$#{$files})
+      {
+        my $template = getOutfileTemplate($fqf_index);
+        next if($template =~ m%n/a%i);
+        foreach my $sample (keys(%$hash),'unmatched')
+          {
+            my $outfile = $template;
+            $outfile =~ s/\%/$sample/;
+            if(-e $outfile)
+              {
+                print STDERR "ERROR: File exists: $outfile.\n";
+                $existing = 1;
+              }
+          }
+      }
+
+    return($existing);
+  }
+
+sub getMultxVersion
+  {
+    my $version = [];
+    my $error   = '';
+    my $is_fqmx = 1;
+
+    my $output  = `($fastq_multx 3>&1 1>&2- 2>&3- ) | head -n 2`;
+    if($?)
+      {
+        $error = "fastq-multx failed with a non-zero exit code [$?]" .
+          ($! =~ /./ ? " and error: $!" : '.') . "\n";
+        return($version,$is_fqmx,$error);
+      }
+    elsif($output !~ /fastq-multx/)
+      {
+        $is_fqmx = 0;
+        return($version,$is_fqmx,$error);
+      }
+    elsif($output =~ /version\D*([\d\.]+)/i)
+      {
+        my $version_string = $1;
+        push(@$version,split(/\./,$version_string));
+      }
+    return($version,$is_fqmx,$error);
+  }
+
+sub fqmxVersionInsufficient
+  {
+    my $used = $_[0];
+    my $reqd = $_[1];
+
+    my $lim =
+      (scalar(@$used) < scalar(@$reqd) ? scalar(@$used) : scalar(@$reqd)) - 1;
+
+    return(1) if($lim == 0);
+
+    my $insufficient = 0;
+    my $num_equal    = 0;
+    foreach my $i (0..$lim)
+      {
+        if($used->[$i] < $reqd->[$i])
+          {
+            $insufficient = 1;
+            last;
+          }
+        elsif($used->[$i] > $reqd->[$i])
+          {last}
+        else
+          {$num_equal++}
+      }
+
+    #Add 1 because num_equal is 1-based and lim is 0-based
+    $lim++;
+
+    #If the required version matches, but the version number of the requirement
+    #is longer than the numbers that were checked
+    if(!$insufficient && $lim == $num_equal && scalar(@$reqd) > $num_equal)
+      {$insufficient = 1}
+
+    return($insufficient);
+  }
+
+sub bcs2fqmxBCFile
+  {
+    my $bcs_file    = $_[0];
+    my $num_indexes = $_[1];
+    my $fqmx_file   = $bcs_file . '.fqmx';
+    my $bchash      = {};
+    my $samplehash  = {unmatched => 'unmatched'};
+    my $uniq_check  = {};
+    my $dupes       = 0;
+
+    unless(open(BCSBCF,$bcs_file))
+      {print STDERR ("ERROR: Unable to open barcode file: $!") && exit(1)}
+
+    unless(open(FQMXBCF,'>',$fqmx_file))
+      {print STDERR ("ERROR: Unable to open fqmx barcode file: $!") && exit(1)}
+
+    my $ln     = 0;
+    while(<BCSBCF>)
+      {
+        $ln++;
+        chomp;
+        next if(/^\s*#/ || /^\s*$/);
+        my @cols = split(/\t/,$_);
```

Comment:
Don't need to use `$_`
```suggestion
        my @cols = split(/\t/);
```